FWCore/Framework/interface/Callback.h

```diff
@@ -223,7 +225,9 @@ namespace edm {
       edm::propagate_const<T*> producer_;
       ESModuleCallingContext callingContext_;
       edm::WaitingTaskList taskList_;
-      method_type method_;
+      // Using std::shared_ptr in order to share the state of the
+      // functor across all clones
+      std::shared_ptr<TFunc> function_;
```

Comment:
I first stored the functor directly as a member variable, but then realized that in general the state of the functor needs to be shared between the clones of the `Callback`.